### PR TITLE
feat: support `no_std` for `reth-execution-errors`

### LINKED ISSUE #8706: execution erorrs crate no-std support
no_std support for
https://github.com/paradigmxyz/reth/blob/da1536f9c4d995c0168bb5af47b2740e5088439f/crates/evm/execution-errors/Cargo.toml#L2-L2

### PR DESCRIPTION
**Pull request by JackG-eth**
Description
no_std support for "reth-execution-errors"

Motivation
closes #8706

### CODE REVIEW

File: crates/evm/execution-errors/src/lib.rs [hunk 1]
```diff
@@ -81,75 +68,206 @@ pub enum BlockValidationError {
...
     ///
     /// [EIP-6110]: https://eips.ethereum.org/EIPS/eip-6110
-    #[error("failed to decode deposit requests from receipts: {0}")]
     DepositRequestDecode(String),
 }
 
+#[cfg(feature = "std")]
+impl std::error::Error for BlockValidationError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        #[allow(deprecated)]
```
Comment (mattsse) [inline crates/evm/execution-errors/src/lib.rs:88]:
```suggestion
```

File: crates/evm/execution-errors/src/lib.rs [hunk 2]
```diff
@@ -81,75 +68,206 @@ pub enum BlockValidationError {
...
+        match self {
+            Self::EVM { error: source, .. } => Some(source),
```
Comment (mattsse) [inline crates/evm/execution-errors/src/lib.rs:90]:
```suggestion
            Self::EVM { error, .. } => Some(source),
```

File: crates/evm/execution-errors/src/lib.rs [hunk 3]
```diff
@@ -81,75 +68,206 @@ pub enum BlockValidationError {
...
+                Some(account_loading_error)
+            }
+            _ => None,
+        }
+    }
+}
+
+impl Display for BlockValidationError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        #[allow(unused_variables, deprecated, clippy::used_underscore_binding)]
```
Comment (mattsse) [inline crates/evm/execution-errors/src/lib.rs:102]:
```suggestion
```

File: crates/evm/execution-errors/src/trie.rs [hunk 4]
```diff
@@ -1,32 +1,86 @@
...
+            Self::StorageRootError(storage_err) => Some(storage_err),
+        }
+    }
+}
+
+impl Display for StateRootError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        match self {
+            Self::DB(db_err) => write!(f, "{}", db_err),
+            Self::StorageRootError(storage_err) => write!(f, "{}", storage_err),
```
Comment (mattsse) [inline crates/evm/execution-errors/src/trie.rs:28]: delegate via err.fmt(f)

File: crates/evm/execution-errors/src/trie.rs [hunk 5]
```diff
@@ -1,32 +1,86 @@
...
+            Self::DB(db_err) => Some(db_err),
+        }
+    }
+}
+
+#[allow(unused_qualifications)]
+impl Display for StorageRootError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        match self {
+            Self::DB(db_err) => write!(f, "{}", db_err),
```
Comment (mattsse) [inline crates/evm/execution-errors/src/trie.rs:77]: error.fmt(f)

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/evm/execution-errors/src/lib.rs [hunk 6]
```diff
@@ -7,71 +7,58 @@
 )]
 #![cfg_attr(not(test), warn(unused_crate_dependencies))]
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
-
+use core::fmt::{Display, Formatter, Result};
```
Comment (mattsse) [inline crates/evm/execution-errors/src/lib.rs:10]:
```suggestion

use core::fmt::{Display, Formatter, Result};
```

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/evm/execution-errors/src/lib.rs [hunk 7]
```diff
@@ -8,70 +8,58 @@
 #![cfg_attr(not(test), warn(unused_crate_dependencies))]
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
```
Comment (mattsse) [inline crates/evm/execution-errors/src/lib.rs:10]: this does not disable std if cfg(std) is not enabled

**Review by mattsse — CHANGES_REQUESTED** [review]
needs actual no-std cfg otherwise lgtm

**Review by onbjerg — APPROVED** [review]
not a huge fan of the cfg for error impl and not using thiserror, but not much that can be done about it i think. when error is stabilized in core maybe thiserror will support nostd and we can revisit. we should make a follow up issue for that, tracking stabilization of error in core

**Review by mattsse — APPROVED** [review]
lgtm
pending @DaniPopes

**Comment by DaniPopes** [comment]
why can we not use thiserror_no_std?

**Comment by DaniPopes** [comment]
You're likely forgetting gating Other(..) variant to also be under `cfg(feature = "std")`, because it cannot be emulated in `no_std`.

**Comment by JackG-eth** [comment]
> You're likely forgetting gating Other(..) variant to also be under `cfg(feature = "std")`, because it cannot be emulated in `no_std`.

This makes a lot of sense, i'll come back to you tonight when I've had a chance to test it.

**Comment by mattsse** [comment]
superseded by #8729

**Pull request closed without merging**